service/matching/db.go

```diff
@@ -729,10 +730,10 @@ func (db *taskQueueDB) emitBacklogGaugesLocked() {
 		return
 	}
 
-	var approximateBacklogCount, totalLag int64
+	var totalLag int64
 	var oldestTime time.Time
 	for _, s := range db.subqueues {
-		approximateBacklogCount += s.ApproximateBacklogCount
+		metrics.ApproximateBacklogCount.With(db.metricsHandler).Record(float64(s.ApproximateBacklogCount), metrics.PriorityTag(locks.Priority(s.Key.Priority)))
```

Comment:
use the same tag as the other matching metrics, `metrics.TaskPriorityTagName`. see other uses of that, e.g. https://github.com/temporalio/temporal/blob/main/service/matching/pri_matcher.go#L447